src/chocolatey/infrastructure.app/services/IRegistryService.cs

```diff
@@ -16,21 +16,43 @@
 
 namespace chocolatey.infrastructure.app.services
 {
+    using System;
     using System.Collections.Generic;
     using domain;
     using Microsoft.Win32;
     using Registry = domain.Registry;
 
     public interface IRegistryService
     {
+        Registry GetInstallerKeys();
+        Registry GetInstallerKeysChanged(Registry before, Registry after);
+        IEnumerable<GenericRegistryValue> GetEnvironmentValues();
+        IEnumerable<GenericRegistryValue> GetNewAndModifiedEnvironmentValues(IEnumerable<GenericRegistryValue> before, IEnumerable<GenericRegistryValue> after);
+        IEnumerable<GenericRegistryValue> GetRemovedEnvironmentValues(IEnumerable<GenericRegistryValue> before, IEnumerable<GenericRegistryValue> after);
+        void SaveRegistrySnapshot(Registry snapshot, string filePath);
+        Registry ReadRegistrySnapshot(string filePath);
+        bool InstallerKeyExists(string keyPath);
+        RegistryKey GetKey(RegistryHive hive, string subKeyPath);
+
+#pragma warning disable IDE1006
+        [Obsolete]

```

Comment:
Is there a reason the Obsolete attributes don't have the additional `This overload is deprecated and will be removed in v3.` in this file?